src/java.desktop/share/classes/javax/swing/filechooser/FileSystemView.java

```diff
@@ -255,6 +256,54 @@ public Icon getSystemIcon(File f) {
         }
     }
 
+   /**
+    * Returns an icon for a file, directory, or folder as it would be displayed
+    * in a system file browser for the requested size.
+    * <p>
+    * Example: <pre>
+    *     FileSystemView fsv = FileSystemView.getFileSystemView();
+    *     Icon icon = fsv.getSystemIcon(new File("application.exe"), 64);
+    *     JLabel label = new JLabel(icon);
+    * </pre>
+    * @implSpec The default implementation gets information from the
+    * {@code ShellFolder} class. Whenever possible, the icon
+    * returned is a multi-resolution icon image,
+    * which allows better support for High DPI environments
+    * with different scaling factors.
+    *
+    * @param f a {@code File} object
+    * @param size width and height of the icon in virtual pixels
```

Comment:
> What are the "virtual pixels"? I remember we refer to something similar by the point in the "user space coordinate system" Or probably we use the virtual pixels somewhere?

It is to say that the sizes are given in the same pixels as other components in the container and are subject to be rendered with different resolution based on the display scaling factor with preserving of relative sizes and proportions. The same terminology is used i.e. in SurfaceData.

Reply:
SurfaceData is not a public class, do we use this term somewhere in the spec? If not then it will be better to use size/points in the user space coordinate system, it is used already in the java2d.

Reply:
If *user coordinate system* is used in similar context in other methods, we should change the wording to match it. I don't mind using a new bug to clarify *virtual pixels*.

Reply:
BTW this is why I recommended filing a CSR after the fix is fully discussed and agreed upon.

Reply:
> BTW this is why I recommended filing a CSR after the fix is fully discussed and agreed upon.

The CSR was filed almost five years ago.